frontend/ui/data/keyboardlayouts/fa_keyboard.lua

```diff
@@ -0,0 +1,126 @@
+    utf8mode_keys = {["🌐"] = true},                      -- The famous globe key for layout switching
+    umlautmode_keys = {["Äéß"] = false},                  -- No need for this keyboard panel
+    keys = {
+        -- first row
+        {  --  1                         2            3      4
+            { dhad,      				dhad,        "~",   "1", },
+            { saad,   					saad,        "`",   "2", },
+            { thaa,      				thaa,        "|",   "3", },
+            { qaf,  					qaf,         "•",   "4", },
+            { fah,                      fah,         "√",   "5", },
+            { ghayn,                    ghayn,       "π",   "6", },
+            { ayin,                     ayin,        "÷",   "7", },
+            { h_aa,                     h_aa,        "×",   "8", },
+            { khaa,                     khaa,        "¶",   "9",  },
+            { ha,                       ha,          "Δ",  "0",  },
+            { jeem,                     jeem,        "‘",   ">"  },
+        },
+        -- second row
+        {  --  1                         2            3       4
+            { sheen,        			sheen,       "£",    _at, },
+            { seen,    					seen,        "¥",    "#", },
+            { yaa,                      yaa,         "$",    "﷼", },
+            { ba,                       ba,          "¢",    "ـ", },
+            { lam,                      lam,         "^",    "&", },
+            { alef,                     alef,        "°",    "-", },
+            { taa,                      taa,         "=",    "+", },
+            { nun,                      nun,         "{",    "(", },
+            { meem,                     meem,        "}",    ")" },
+            { kaf,                      kaf,         "\\",   "٫", },
+            { gaf,                      gaf,         "/",     "<", },
+        },
+        -- third row
+        {  --  1                         2             3       4
+            { th_aa,       			    th_aa,        "٪",    "/", },
+            { tah,                      tah,          "©",     "«", },
+            { jee,                      jee,          "®",    "»", },
+            { zay,                      zay,          "™",    ":", },
+            { raa,                      raa,          "✓",   "؛", },
+            { thaal,          			thaal,        "[",    "!", },
+            { daal,                     daal,         "]",   "؟", },
+            { pe,                       pe,         "↑",   "↑", },
+            { waw,             			waw,          "←",    "←", },
+            { che,                      che,        "→",   "→",  },
+            { label = "",
+              width = 1,
+              bold = false
+            },
+        },
+        -- fourth row
+        {
+            {"نشانه‌ها","نشانه‌ها","الفبا","الفبا",
+              width = 1},
```

Comment:
It seems that you should adjust the width values slightly? (E.g., this one `2`, spacebar down to `1.5`, or something.)
![Screenshot_2021-01-31_20-48-14](https://user-images.githubusercontent.com/202757/106396060-f56b5c00-6405-11eb-8ace-d8ea83fc85b1.png)

Reply:
i preffer to have a big space key but for this situation i have multiple option
1. koreader automatically change the font of size of label in this situation (complicated)
2. use a small icon instead of "نشانه‌ها" and "حرکات" for smaller key size
3. change size that mean small 'Enter' key [1.1] and some other adjustment
4. combine space key with Zero-width non-joiner key and don't change other key size

i need more time to test all of this possibility to make a reasonable disposition.
what do you think? 
i will fix this as soon as possible

Reply:
i try this with ba0f834d3d6c9dd0bf36139348db51f87ef548ab, 362f2b6ec2fb68c826c37f6767d72707c0430357 and bf9f9e5f0ec655bf8f2f216ec031c931b3efa146 please report if it has any problem